### PR TITLE
Register user 2

### PR DESCRIPTION
**Pull request by MihaliGabriel**
Added register endpoint and login endpoint. Changed the model for the User , added validations.

### CODE REVIEW

**Review by sotanioachim — APPROVED** [review]
Bravo Gabi! Nota 10, ai o bere!

File: Social_Platform_Backend/src/main/java/com/example/social_platform_backend/Controller/GlobalExceptionHandler.java
Comment (alextiugan15) [inline Social_Platform_Backend/src/main/java/com/example/social_platform_backend/Controller/GlobalExceptionHandler.java:1]: da comentari bravo gabe

File: Social_Platform_Backend/src/main/java/com/example/social_platform_backend/Facade/LoginDto.java
Comment (alextiugan15) [inline Social_Platform_Backend/src/main/java/com/example/social_platform_backend/Facade/LoginDto.java:1]: LoginDTO make an issue branch to solve this for naming conventions

File: Social_Platform_Backend/src/main/java/com/example/social_platform_backend/Facade/RegisterDto.java
Comment (alextiugan15) [inline Social_Platform_Backend/src/main/java/com/example/social_platform_backend/Facade/RegisterDto.java:1]: same naming convention...

**Review by AlexX15-T — APPROVED** [review]
bravo gabi, se poate si pe natural 100%

**Pull request merged**